hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -308,7 +312,9 @@ public static boolean hasEnoughSpace(DatanodeDetails datanodeDetails,
 
     if (dataSizeRequired > 0) {
       for (StorageReportProto reportProto : datanodeInfo.getStorageReports()) {
-        if (reportProto.getRemaining() > dataSizeRequired) {
+        if (hasVolumeEnoughSpace(reportProto.getCapacity(),
+            reportProto.getRemaining(), reportProto.getCommitted(),
+            dataSizeRequired, conf)) {
```

Comment:
The configuration used in `getMinVolumeFreeSpace` is the Datanode configuration, what if the configuration is different for different Datanodes or not available in SCM?
Maybe `volumeFreeSpaceToSpare` should also report